Domybest (#1)
Kosuke-iga/SSS-master-thesis: doc/2_related.tex

\chapter{Related Work}\markright{Chapter 2: Related Work}\label{sec:related_work} 
In order to detect IoT malwares, several schemes have been proposed. 
These schemes are divided into router-side detection and device-side detection.
Router-side detection is performed at a gateway on the basis of the features regarding propagation behavior for building a botnet.
Meanwhile, device-side detection is conducted at each IoT device by utilizing static features extracted from malware itself.
The representative schemes are explained in the following subsections.

\section{Router-side detection}
Router-side detctions leverage the fact that there exists strong evidence of malwares in network traffic.

> Kumar et al. \cite{router} propose the scheme focusing on network traffic while the scanning and propagation phase in particular which is observed in most of the existing malware behaviour for the purpose of building a botnet. 

Reviewer: ```suggestion
Kumar et al. \cite{router} propose the scheme focusing on network traffic during the scanning and propagation phase in particular which is observed in most of the existing malware behavior for the purpose of building a botnet. 

```

> That scheme detect abnormal traffic caused while this phase which cannot be seen in benign cases utilizing machine learning.

Reviewer: ```suggestion
That scheme detect abnormal traffic caused during this phase which cannot be seen in benign cases utilizing machine learning.

```

> It utilizes the number of unique destination IP addresses and packets per destination IP addresses as features observing incoming packets which have spcific target destination port number that the existing malwares tend to send.

Reviewer: ```suggestion
It utilizes the number of unique destination IP addresses and packets per destination IP addresses as features for observing incoming packets that have specific target destination port numbers used by frequently existing malware samples.
```
こういうこと？？

> The number of unique destination IP addresses in case of malware-induced scanning traffic will be far more than benign traffic since the malware generate random IP addresses and send malicious requests to them.

Reviewer: ```suggestion
The number of unique destination IP addresses in case of malware-induced scanning traffic will be far more than benign traffic since the malware generate random IP addresses and sends malicious requests to them.

```

> Also, the number of packets per destination IP addresses seeks to exploit the fact that malware typically do not send mulitiple malicious packets to the same IP addresses (only a single packet is sent in most cases), possibly to cover as many devices as possible during the scanning/propagation phase.

Reviewer: ```suggestion
Also, using the number of packets per destination IP addresses is instrumental in exploiting the fact that malware samples typically do not send multiple malicious packets to the same IP addresses (only a single packet is sent in most cases), possibly to cover as many devices as possible during the scanning/propagation phase.

```
こんな感じ？
seek to がよくわからなかった。

> By being based on the scanning/propagation phase, it can detect and isolate the bots before they can participate in an actual attack such as DDoS.

Reviewer: ```suggestion
On the basis of these tendencies during the scanning/propagation phase, it can detect and isolate the bots before they can participate in an actual attack such as DDoS.

```

However, that scheme is subject to local network attacks where IoT devices are targeted directly by adversaries in the same local network.
Thus, exploring the malware detection schemes which can be realized by each IoT device is needed due to the vulnerability against local network attacks.

\section{Device-side detection} 

> Device-side detection schemes are conducted by utilizing the features extracted from malware file itself at each device on the basis of malwares are tend to have similality among them since adversaries make them based on the existing ones.

Reviewer: ```suggestion
Device-side detection schemes are conducted by utilizing the features extracted from a malware sample itself at each device.
This is because adversaries tend to make new malware samples on the basis of existing ones in many cases, which results in high similarities among them.

```
